tests/test_main.py

```diff
@@ -0,0 +1,21 @@
+from hugie.__main__ import app
+from typer.testing import CliRunner
+
+runner = CliRunner()
+
+
+def test_ui_command(monkeypatch):
+    def mock_open(url):
```

Comment:
better to define the `mock_open` as a `Mock` https://docs.python.org/3/library/unittest.mock.html#the-mock-class and then test it is called `assert_called` https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called